### PR TITLE
OCPCLOUD-898 - Use klog/controller runtime-logger in mao

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
With both `klog` and `glog` initialized in different controllers, using global variables defined in one of them, and used in another, becomes impossible, as the command-line `(k/g)log` flags are internally redefined by at least one of the libraries.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
@Danil-Grigorev Please rebase this PR and a description to explain the motivation behind this change

**Comment by Danil-Grigorev** [comment]
@JoelSpeed Just did a rebase and updated the description with additional motivation.

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,6 @@ go 1.13
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/evanphx/json-patch v4.5.0+incompatible
-	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
```
Comment (JoelSpeed) [inline go.mod:8]: Would expect to see some go.sum update as well, have you run `make vendor`?

Comment (Danil-Grigorev) [reply go.mod:8]: Let me check. Sometimes I need to remove `go.sum` so the vendor takes effect.

Comment (JoelSpeed) [reply go.mod:8]: > Sometimes I need to remove go.sum so the vendor takes effect.

You should never need to do that... What errors are you getting that make you want to do that?

File: pkg/operator/sync.go [hunk 2]
```diff
@@ -59,12 +59,12 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
...
 			// Just log the error here.  We still want to
 			// return the outer error.
-			glog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
+			klog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
 		}
-		glog.Errorf("Error syncing machine API webhook configurations: %v", err)
+		klog.Errorf("Error syncing machine API webhook configurations: %v", err)
 		return err
 	}
-	glog.V(3).Info("Synced up all machine API webhook configurations")
```
Comment (JoelSpeed) [inline pkg/operator/sync.go:67]: I think this file change ended up in the wrong commit

**Comment by enxebre** [comment]
@Danil-Grigorev this needs rebase

File: go.mod [hunk 3]
```diff
@@ -26,6 +25,7 @@ require (
 	k8s.io/apimachinery v0.19.0
 	k8s.io/client-go v0.19.0
 	k8s.io/code-generator v0.19.0
+	k8s.io/klog v1.0.0
```
Comment (JoelSpeed) [inline go.mod:28]: We want to only use `klog/v2` now, just need to replace all imports `k8s.io/klog` with `k8s.io/klog/v2`, can you ensure this is done?

**Review by JoelSpeed — COMMENTED** [review]
/approve

Thanks @Danil-Grigorev

File: cmd/machine-api-operator/start.go [hunk 4]
```diff
@@ -48,22 +49,25 @@ func init() {
 	rootCmd.AddCommand(startCmd)
 	startCmd.PersistentFlags().StringVar(&startOpts.kubeconfig, "kubeconfig", "", "Kubeconfig file to access a remote cluster (testing only)")
 	startCmd.PersistentFlags().StringVar(&startOpts.imagesFile, "images-json", "", "images.json file for MAO.")
+
+	klog.InitFlags(nil)
+	flag.Parse()
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
```
Comment (JoelSpeed) [inline cmd/machine-api-operator/start.go:55]: Why are we adding pflag here? Should we not just be using the flag package?

Comment (Danil-Grigorev) [reply cmd/machine-api-operator/start.go:55]: No, apparently cobra overrides the flags and needs to pass klog defaults this way. This was the cause for the CI failures

**Comment by Danil-Grigorev** [comment]
@elmiko @JoelSpeed @alexander-demichev PTAL

**Review by alexander-demichev — APPROVED** [review]
/lgtm

**Pull request merged**